### ISSUE TITLE
Mad Max (234140)

### ISSUE THREAD

**Issue by AlexP11223** [ValveSoftware/Proton#4170]
# Compatibility Report
- Name of the game with compatibility issues: **Mad Max**
- Steam AppID of the game: **234140**

## System Information
- GPU: **GTX 1080**
- Driver/LLVM version: **nvidia 450.66**
- Kernel version: **5.4**
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/AlexP11223/7506e04f1619d361761b1e0c4441e96b
- Proton version: **5.0.9, 5.0.10 RC**

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms && Reproduction

Playing Mad Max with Proton as suggested in ProtonDB (the native version from Feral has worse performance, no 144 Hz, etc.). Everything is fine, 140+ fps.
Except these weird crashes.

It crashes if I start the game and drive from a stronghold to any "distant" point. Always in the same place.
For example:

![](https://i.imgur.com/ZDvZzg7.png)

If I go somewhere near the stronghold first, complete some activities nearby, then later everything is fine and I can drive anywhere. So it crashes only if I go too far away too quickly after loading the game :D

In Proton log I found only this at the end:

```
...
679.609:023c:0240:trace:seh:execute_cfa_instructions 7f85d89ea264: DW_CFA_def_cfa %rsp, 8
679.609:023c:0240:trace:seh:execute_cfa_instructions 7f85d89ea264: DW_CFA_advance_loc 1
679.609:023c:0240:trace:seh:execute_cfa_instructions 7f85d89ea265: DW_CFA_restore_state
679.609:023c:0240:err:seh:setup_exception stack overflow 1632 bytes in thread 0240 eip 00007f85db837228 esp 0000000000120fb0 stack 0x120000-0x121000-0x220000
680.708:0058:0068:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\drivers\\winehid.sys" : builtin
680.708:0058:0068:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\hidclass.sys" : builtin
680.710:0058:0068:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\drivers\\winebus.sys" : builtin
pid 9175 != 9174, skipping destruction (fork without exec?)
```

Full log: https://drive.google.com/uc?export=download&id=1IlIokJcYy7oN6YO580kXh8uOqNEW3CN3

The native version doesn't have this issue.

**Comment by GloriousEggroll** [comment]
have you tried the native version with the vulkan renderer?

https://www.feralinteractive.com/en/news/753/

**Comment by AlexP11223** [comment]
> have you tried the native version with the vulkan renderer?

Yeah, but it only improves performance, other issues of native version still remain, such as no 144 Hz. And with vulkan renderer it also crashes on Alt+Tab with `vkWaitForFences failed: -4` error.

**Comment by GloriousEggroll** [comment]
144hz native vulkan beta works fine here. the screen resolution options say @60 but it does indeed go to 144hz, both windows borderless and fullscreen. My monitor is 144hz and verified set at 144hz in display settings. 
Alt+tab also works here on KDE. I used it when taking the screenshots, however I am also on AMD with RADV. I can test on my other machine which has a 1660 in it on manjaro with the same drivers and provide another update.

![madmax2](https://user-images.githubusercontent.com/11287837/92310441-8cbda780-ef6b-11ea-8f85-3b5ad0630aaf.png)
![madmax](https://user-images.githubusercontent.com/11287837/92310442-8d563e00-ef6b-11ea-8b2e-db0a02ef970a.png)

**Comment by AlexP11223** [comment]
> both windows borderless and fullscreen

hm, borderless? I think it's available only in Proton, in native (both standard and vulkan) I have only fullscreen on/off (windowed).

**Comment by GloriousEggroll** [comment]
> > both windows borderless and fullscreen
> 
> hm, borderless? I think it's available only in Proton, in native (both standard and vulkan) I have only fullscreen on/off (windowed).

Sorry for the confusion -- i meant I tried it in native both windowed without border  (in kwin you can remove the border) and fullscreen. Both ways worked for 144hz on AMD

**Comment by AlexP11223** [comment]
tried native vulkan again:

1. Launched in fullscreen, Alt+Tab --> crash
2. Launched, switched to windowed --> crash during switching
3. Launched in windowed --> no crashes during Alt+Tab even after switching to fullscreen :thinking: 

I think I will stick to the standard native for now, because the performance seems to be good enough so far (I guess because I have top hardware that came out several years after the game release :D) + maybe decreasing some settings like shadows could help, currently everything is at max, 1440p., 60-110 fps.
